Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -516,6 +512,26 @@ LocalFrameView* AsyncScrollingCoordinator::frameViewForScrollingNode(ScrollingNo
     }
 
     return nullptr;
+
+}
+
+LocalFrameView* AsyncScrollingCoordinator::frameViewForScrollingNode(ScrollingNodeID scrollingNodeID) const
+{
+    if (!m_scrollingStateTree->rootStateNode())
+        return nullptr;
+
+    RefPtr localMainFrame = dynamicDowncast<LocalFrame>(page()->mainFrame());
+    if (localMainFrame)
+        return frameViewForScrollingNode(*localMainFrame, scrollingNodeID);
+    if (!localMainFrame && is<RemoteFrame>(page()->mainFrame())) {
```

Comment:
All this is probably unnecessary.  With site isolation off, and in the main frame process, you can count on Page::rootFrames() to return the main frame.  If the main frame is local and there's also a local grandchild frame that is also a root frame, then this code is actually incorrect.

Reply:
Right fixed.